src/main/pages/che-7/hands-on-guidance/proc_creating-a-workspace-from-code-sample.adoc

```diff
@@ -1,27 +1,125 @@
 [id="creating-a-workspace-from-code-sample_{context}"]
 = Creating a workspace from code sample
 
-This paragraph is the procedure module introduction: a short description of the procedure.
+This section will show how to select a stack from the User Dashboard and start a workspace with its sample code.
+
+It will also explain how to customize a workspace using the User Dashboard.
+
+[NOTE]
+====
+Every new devfile based stack should link:https://github.com/eclipse/che/issues/13529[come with one sample codebase].
+====
 
 [discrete]
 == Prerequisites
 
-* A bulleted list of conditions that must be satisfied before the user starts following this assembly.
-* You can also link to other modules or assemblies the user must follow before starting this assembly.
-* Delete the section title and bullets if the assembly has no prerequisites.
+* You need an OKD, OCP, or OSD cluster, if using Che on Openshift. See below for use of link:https://www.okd.io/minishift/[Minishift].
+* Or, you need a Kubernetes cluster such as link:https://github.com/kubernetes/minikube#installation[Minikube] to use Che on Kubernetes.
 
 [discrete]
 == Procedure
 
-. Start each step with an active verb.
+=== Accessing Che User Dashboard
+
+The dashboard is accessible on your cluster from a URL like http://`che-instance`.`your.IP.address.here`.nip.io/dashboard/.
+
+==== Minishift
+
+If you are using link:https://www.okd.io/minishift/[Minishift], you can use the Che add-on to automatically deploy Che to your Minishift (OKD) instance.
+
+[NOTE]
+====
+For versions of link:https://github.com/minishift/minishift/releases[Minishift up to 1.34], you need to apply a small change:
+
+```
+minishift --profile che addons apply --addon-env PLUGIN__REGISTRY__URL=https://che-plugin-registry.openshift.io/v3 che
+```
+====
+
+. Next, log into your console at https://`your.IP.address.here`:8443/console/.
+
+. From there, select the mini-che project:
+
+* https://`your.IP.address.here`:8443/console/project/mini-che/overview
+
+. Then click through to the mini-che route link, which will load the dashboard by default and prompt you to create a workspace.
+
+* http://che-mini-che.`your.IP.address.here`.nip.io/
+
+==== OCP and OSD
+
+If using OCP or OSD, talk to your administrator to obtain the URL for the che instance.
+
+=== Selecting a Stack
+
+Choose a stack from the list. Note that only stacks marked with Che 7 are fully compatible with Che 7 server, but the older Che 6 based ones may still work.
```

Comment:
Only mention of Che 6 is now this line:

```
. From the *Editors* tab, you can enable/disable workspace editors. For example, if you wanted to switch to the Che 6 editor, you could select the GWT IDE instead of the Che 7 editor, based on Theia.
```